Muszilla/src/Controller/HomeController.php

```diff
@@ -0,0 +1,309 @@
+        return $this->render('Muszilla/error.html.twig', ['error' => $error, 'user' => $user]);
+      }
+      
+    }
+    return $this->render('Muszilla/login.html.twig');
+  }
+
+
+  #[Route('/', name: 'app_root')]
+  /**
+   * Root route to handle the condition when the user tries to go back to the 
+   * login page after logging in.
+   *
+   * @return Response
+   *    Returns the home page if the user is already logged in, otherwise 
+   *    redirects the user to the home page.
+   */
+  public function root(): Response {
+    session_start();
+    if(isset($_SESSION['email'])) {
+      $user = $this->user->getUserData($_SESSION['email']);
+      return $this->render('Muszilla/home.html.twig', ['user'=>$user]);
+    }
+    return $this->render('Muszilla/login.html.twig');
+  }
+
+  #[Route('/logout', name: 'app_logout')] 
+  /**
+   * Function the destroy the user session and redirect to the login page if the
+   * user wants to log out.
+   *
+   * @return Response
+   *    If the user is logged in, then destroys the user session and redirects
+   *    to to the login page, otherwise if not logged in 
+   *    redirects to the login page.
+   */
+  public function logout(): Response {
+    session_start();
+    if(isset($_SESSION)) {
+      session_unset();
+      session_destroy();
+      return $this->redirectToRoute('app_login');
+    }
+    return $this->redirectToRoute('app_login');
+  }
+
+  #[Route('/register', name: "app_register")] 
+  /**
+   * Function to render the register page.
+   *
+   * @param Request $request
+   *    Stores the credentials entered by the user.
+   * 
+   * @return Response
+   *    Renders the registration page if the user is not logged in, and asks 
+   *    the user to register otherwise, render the home page, if already looged
+   *    in.
+   */
+  public function register(Request $request):Response {
+    session_start();
+    if(isset($_SESSION['email'])) {
+      $user = $this->user->getUserData($_SESSION['email']);
+      return $this->redirectToRoute('app_root', ['user'=>$user]);
+    }
+    return $this->render('Muszilla/register.html.twig');
+  }
+
+  #[Route('/home', name: 'app_home')]
+  /**
+   * Function to render the home page to the user, if the user is logged in, and
+   * is a valid user.
+   *
+   * @param Request $request
+   *    Stores the credentials provided by the user to log into his account.
+   * 
+   * @return Response
+   *    Renders the home page if the user is logges in, and the credentials are
+   *    correct, otherwise renders the error page with a suitable error message.
+   *    Renders the login page, if the user is not logged in and tries to go to 
+   *    the home page. 
+   */
+  public function home(Request $request):Response {
+    session_start();
+
+    // Handles the condition where the user tries to log in.
+    if(!empty($request->request->all())) {
+      $user = $request->request->all();
+      $user_exist = $this->user->login($user);
```

Comment:
Use camle case.

Reply:
Variables should be named using lowercase, and words should be separated either with uppercase characters (example: $lowerCamelCase) or with an underscore (example: $snake_case). 
From Drupal.org.